### PR TITLE
Charles/caption float

### PR DESCRIPTION
**Pull request by chaznied**
## JIRA Ticket
[https://tbnextjs.atlassian.net/browse/NJP-2114]

## Describe Changes in this Pull Request

- Updated caption styles in fullscreen for video/images/daf images

## Testing Instructions

- Pull this from the tollbrothers-ui repo [https://github.com/tollbros/tollbrothers-ui/tree/charles/captionFloat]
- Run `npm run start`
- Pull this from the tollbrothers.com repo [https://bitbucket.org/tollbrothers/com.tollbrothers.www/branch/charles/NJP-2114-tollbrothers-ui-advantage-page-]
- Update `package.json` with `"@tollbrothers/tollbrothers-ui": "file:../tollbrothers-ui",`
- Run `npm i`
- Run `npm run dev`
- Test here - [http://localhost:3000/advantage]
- Test Daf images here - [http://localhost:3000/luxury-homes-for-sale/California/Skylar-by-Toll-Brothers/Quick-Move-In/261420]
- Test randomly here - [http://localhost:3000/luxury-homes-for-sale/New-Jersey/Canter-Square/Mueller]

## Screen Shot(s)/Video(s) of Working Changes
Please add screen shots or screen recordings showing the changes

### Tasks:

1. [ ] Changes Meet JIRA Tickets Requirements
2. [ ] Commit message with `fix` or `feat` to trigger new release
3. [ ] Tested changes on Mobile, Tablet and Desktop screen sizes
4. [ ] Tested changes on Chrome, FireFox and Safari
5. [ ] Demoed to Design Team (For new features only)
6. [ ] Notified Digital Marketing Team if changes will impact A/B Tests
7. [ ] Checked Tracking/Analytics (Notified Digital Marketing Team if changes impact tracking)

### CODE REVIEW

**Comment by cormdog** [comment]
Needs a little tweaking on mobile. Screenshot attached.
IMG_0564

File: src/components/FullScreenGallery.module.scss [hunk 1]
```diff
@@ -46,7 +46,8 @@
         width: 100%;
         height: 100%;
         margin: auto;
-        object-fit: contain;
+        object-fit: cover;
```
Comment (mikeduarte) [inline src/components/FullScreenGallery.module.scss]: This causes a really big change to the tb.com gallery, especially on mobile. We might want to get input from Justin.

Before:
Screenshot 2024-05-15 at 4 32 16 PM


Your Branch:
Screenshot 2024-05-15 at 4 32 09 PM

File: src/components/GalleryMedia.js [hunk 2]
```diff
@@ -15,7 +15,9 @@ function GalleryMedia({
   index,
   mediaCount,
   onLoad,
-  backgroundColor
+  backgroundColor,
+  isDesignerAppointed
```
Comment (mikeduarte) [inline src/components/GalleryMedia.js]: I don't think this is being use.

File: src/components/FullScreenGallery.js [hunk 3]
```diff
@@ -29,7 +29,8 @@ export const FullScreenGallery = ({
   onPrevious = () => {},
   initialSlide = 1,
   backgroundColor,
-  portalId
+  portalId,
+  isDesignerAppointed
```
Comment (mikeduarte) [inline src/components/FullScreenGallery.js]: Similar to the other PR, we might want to pass a classes prop from the parent app to override styles here. We can huddle about it.

**Review by mikeduarte — CHANGES_REQUESTED** [review]
Left a few comments

**Comment by chaznied** [comment]
@mikeduarte  - updated, only adjusting captions for videos/walkthrough above 768 and added class prop

**Comment by Chenzo** [comment]
Hey, per our conversation, fix this from floating over the controls (via Justin's fix of removing all that from any video)

Screenshot 2024-05-28 at 11 35 43 AM

**Review by Chenzo — CHANGES_REQUESTED** [review]
see my previous comment

**Comment by chaznied** [comment]
@Chenzo , @mikeduarte  - I just remembered we can't remove captions from any video because of the disclaimer i had to add for all inside maps...

**Comment by chaznied** [comment]
@Chenzo , @mikeduarte  - Updated GalleryMedia.js to only show captions for images and insidemaps walkthroughs and removing counter and social from videos and walkthroughs

File: src/components/GalleryMedia.js [hunk 4]
```diff
@@ -25,20 +26,25 @@ function GalleryMedia({
...
   let type = 'image'
+  let customClass = null
   const [showMedia, setShowMedia] = useState(false)
   const imgRef = useRef()
-  let imgCount = (index + 1).toString() + " / " + mediaCount.toString();
+  const imgCount = (index + 1).toString() + ' / ' + mediaCount.toString()
   if (media?.type?.includes('video')) {
     modelLink = null
     type = 'video'
+    customClass = styles.mediaVideo
```
Comment (mikeduarte) [inline src/components/GalleryMedia.js]: Do we need this if we aren't showing captions on videos?

File: src/components/FullScreenGallery.module.scss [hunk 5]
```diff
@@ -47,6 +47,13 @@
         height: 100%;
         margin: auto;
         object-fit: contain;
+        //padding-bottom: 50px;
+      }
+
+      iframe {
+        @include breakpoint(sm) {
+          object-fit: cover;
```
Comment (mikeduarte) [inline src/components/FullScreenGallery.module.scss]: I don't think we need this. I don't see a difference when I comment it out.

File: src/components/GalleryMedia.js [hunk 6]
```diff
@@ -25,20 +26,25 @@ function GalleryMedia({
...
     iframeSrc = getVideoURL(media)
   } else if (media?.type?.includes('walkthrough')) {
     modelLink = null
     type = 'walkthrough'
     iframeSrc = getWalkthroughURL(media)
+    customClass = styles.mediaVideo
```
Comment (mikeduarte) [inline src/components/GalleryMedia.js]: Maybe this should only get applied if there is a caption and showCaption is true?

**Review by mikeduarte — CHANGES_REQUESTED** [review]

**Review by mikeduarte — APPROVED** [review]

**Comment by chaznied** [comment]
@Chenzo  - I kept the caption for inside maps (legal) and walkthroughs (Justin) and images, removed captions from videos (Justin), removed social/counter from video and walkthroughs (Justin). Images remain the same

**Comment by Chenzo** [comment]
Matterports don't seem to work. We chatted. Waiting on fix.

**Review by Chenzo — APPROVED** [review]
It's totally something from Matterport in either your cookies, session or local storage.

**Pull request merged**